packages/@ember/-internals/glimmer/lib/helpers/each-in.ts

```diff
@@ -15,6 +15,9 @@ import { consumeTag } from '@glimmer/validator';
   The default behavior of `{{#each}}` is to yield its inner block once for every
   item in an array passing the item as the first block parameter.
 
+  For performance reasons we are using a standard `for` loop for `{{each}}` iteration,
+  what's why sparse arrays is not supported with `{{each}}`, but supported with `{{each-in}}`.
```

Comment:
FWIW, I don't know that "for performance reasons" is a good thing to say here.

Also, have we confirmed this is still true? There have been some quasi-recent (since 2016 which is when https://github.com/emberjs/ember.js/issues/14725#issuecomment-269389814 was) changes that will attempt to leverage native iterables if possible. In that case, I _think_ sparse arrays are handled more as expected? We should test either way to confirm.

---

If we do need to mention that sparse arrays are unsupported, we should just say that (and not compare to `each-in` or mention performance).

Reply:
@pzuraq could you clarify it or suggest better wording?